### PR TITLE
Add cacert param for self signed spacewalk usage

### LINKED ISSUE #29649: feature request: rhn_channel, and other rhn_* modules, should have validate_certs parameter
From @snemir2 on 2016-09-15T20:21:40Z

##### ISSUE TYPE

- Feature Idea
##### COMPONENT NAME

rhn_channel
rhn_*
##### ANSIBLE VERSION

```

[snemirovsky@SN-WS-Fedora24 patch]$ ansible --version
[WARNING]: log file at /var/log/ansible.log is not writeable and we cannot create it, aborting

ansible 2.1.1.0
  config file = /ansible/conf/ansible.cfg
  configured module search path = Default w/o overrides

```
##### STEPS TO REPRODUCE

```

```

##### EXPECTED RESULTS

excected my code to work 
##### ACTUAL RESULTS

```
TASK [RHN_CHANNEL_configure : Configure Channel on Satellite] ******************
An exception occurred during task execution. To see the full traceback, use -vvv. The error was: ssl.SSLError: [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed (_ssl.c:590)
```

Copied from original issue: ansible/ansible-modules-core#4876

### PR DESCRIPTION
**Pull request by gali0**
##### SUMMARY

rhn_channel raise a CERTIFICATE_VERIFY_FAILED error when use with self signed certificate

Fixes  #29649

##### ISSUE TYPE

- Feature Pull Request

##### COMPONENT NAME

rhn_channel

##### ADDITIONAL INFORMATION

Error raised before change:
```paste below
TASK [RHN_CHANNEL_configure : Configure Channel on Satellite] ******************
An exception occurred during task execution. To see the full traceback, use -vvv. The error was: ssl.SSLError: [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed (_ssl.c:590)
```

### CODE REVIEW

File: lib/ansible/modules/packaging/os/rhn_channel.py [hunk 1]
```diff
@@ -47,6 +49,11 @@
         description:
             - RHN/Satellite password.
         required: true
+    cacert:
```
Comment (jamescassell) [inline lib/ansible/modules/packaging/os/rhn_channel.py]:
```suggestion
    ca_cert:
```

**Review by jamescassell — CHANGES_REQUESTED** [review]
See https://github.com/ansible/ansible/issues/52506

**Comment by gali0** [comment]
> 
> 
> See #52506

You're right, thanks

File: lib/ansible/modules/packaging/os/rhn_channel.py [hunk 2]
```diff
@@ -115,7 +127,15 @@ def main():
     password = module.params['password']
 
     # initialize connection
-    client = xmlrpc_client.ServerProxy(saturl)
+    if module.params['ca_cert'] is not None and sys.version_info > (2, 7, 9):
+        ctx = ssl.create_default_context()
+        ctx.load_verify_locations(cafile=module.params['ca_cert'])
+        client = xmlrpc_client.ServerProxy(saturl, context=ctx)
+    else:
+        if sys.version_info < (2, 7, 9):
```
Comment (felixfontein) [inline lib/ansible/modules/packaging/os/rhn_channel.py]:
```suggestion
        if module.params['ca_cert'] and sys.version_info < (2, 7, 9):
```

File: lib/ansible/modules/packaging/os/rhn_channel.py [hunk 3]
```diff
@@ -115,7 +127,15 @@ def main():
     password = module.params['password']
 
     # initialize connection
-    client = xmlrpc_client.ServerProxy(saturl)
+    if module.params['ca_cert'] is not None and sys.version_info > (2, 7, 9):
```
Comment (felixfontein) [inline lib/ansible/modules/packaging/os/rhn_channel.py]: If your claim below is correct, you need:
```suggestion
    if module.params['ca_cert'] is not None and sys.version_info >= (2, 7, 9):
```

**Review by felixfontein — COMMENTED** [review]
You need a [changelog fragment](https://docs.ansible.com/ansible/latest/community/development_process.html#changelogs-how-to).

**Comment by gali0** [comment]
> 
> 
> You need a [changelog fragment](https://docs.ansible.com/ansible/latest/community/development_process.html#changelogs-how-to).

Done

Thanks.

**Review by felixfontein — APPROVED** [review]

**Comment by yakhatape** [comment]
any update about this PR ? I'm interested on it too :)

**Comment by gali0** [comment]
bot_status

**Comment by gali0** [comment]
> 
> 
> any update about this PR ? I'm interested on it too :)

All community plugins have moved to another repos, but I can't find any infomations on how to move this PR to this new repo, but I'm still working on getting it accepted :)

**Comment by felixfontein** [comment]
Most current info is on https://github.com/ansible-collections/overview/, but please note that currently everything is still in flux and community.general (where this has been moved) is still not completely ready yet.

**Comment by gali0** [comment]
> 
> 
> Most current info is on https://github.com/ansible-collections/overview/, but please note that currently everything is still in flux and community.general (where this has been moved) is still not completely ready yet.

Okay, thanks for your answer, so I'll wait until you're ready.

**Comment by yakhatape** [comment]
many thanks for your update @gali0 @felixfontein  ;)

**Pull request closed without merging**